packages/core/test/transaction.spec.ts

```diff
@@ -48,4 +48,34 @@ describe('Transaction', () => {
             expect(transaction.locktime).toBe(tx.raw.locktime);
         },
     );
+
+    const complexTransactions = [
+        'P2PK',
+        'P2SH P2PK',
+        'P2SH P2PKH',
+        'P2WSH P2PK',
+        'P2WSH P2PKH',
+        'P2SH P2WSH P2PKH',
+        'P2SH P2WSH P2PK',
+        'Standard transaction (14:2)',
+        'Multisig',
+        'P2WSH Multisig',
+        'P2SH P2WSH Multisig',
+        'Coinbase transaction w/ witness',
+        'P2SH Multisig',
+        'Coinbase transaction',
+    ];
+
+    it.each(
+        validTransactions.filter(
```

Comment:
why do we need to do this filtering? shouldn't `getPublicKeyFromInput()` return `null` for all cases that are being filtered out?

Reply:
That would mean I have to add `null` to fixtures as the `inPubkeys` which seemed wrong to me as it could be a P2SH tx or something which has a PK. It's just that currently we only support a few txs, and the null cases should not be tested against IMO. What do you think?

Reply:
fair point